src/analysis/normed_space/spectrum.lean

```diff
@@ -131,6 +140,125 @@ end
+    { rw [units.smul_def, ←algebra.algebra_map_eq_smul_one, ←mem_resolvent_set_iff],
+      refine mem_resolvent_set_of_spectral_radius_lt _,
+      rwa [units.coe_inv', normed_field.nnnorm_inv, coe_inv (nnnorm_ne_zero_iff.mpr
+        (units.coe_mk0 hz ▸ hz : (u : 𝕜) ≠ 0)), lt_inv_iff_lt_inv] } }
+end
+
+/-- In a Banach algebra `A` over `𝕜`, for `a : A` the function `λ z, (1 - z • a)⁻¹` is
+differentiable on any closed ball centered at zero of radius `r < (spectral_radius 𝕜 a)⁻¹`. -/
+theorem inverse_one_sub_smul_differentiable_on [complete_space A] {a : A} {r : ℝ≥0}
+  (hr : (r : ℝ≥0∞) < (spectral_radius 𝕜 a)⁻¹) :
+  differentiable_on 𝕜 (λ z : 𝕜, ring.inverse (1 - z • a)) (metric.closed_ball 0 r) :=
+begin
+  intros z z_mem,
+  apply differentiable_at.differentiable_within_at,
+  have hu : is_unit (1 - z • a),
+  { refine is_unit_one_sub_smul_of_lt_inv_radius (lt_of_le_of_lt (coe_mono _) hr),
+    simpa only [norm_to_nnreal, real.to_nnreal_coe]
+      using real.to_nnreal_mono (mem_closed_ball_zero_iff.mp z_mem) },
+  have H₁ : differentiable 𝕜 (λ w : 𝕜, 1 - w • a) := (differentiable_id.smul_const a).const_sub 1,
+  exact differentiable_at.comp z (differentiable_at_inverse hu.unit) (H₁.differentiable_at),
+end
+
+end one_sub_smul
+
+section gelfand_formula
+
+open filter ennreal continuous_multilinear_map
+open_locale topological_space
+
+/- the assumption below that `A` be second countable is a technical limitation due to
+the current implementation of Bochner integrals in mathlib. Once this is changed, we
+will be able to remove that hypothesis. -/
+variables
+[normed_ring A] [normed_algebra ℂ A] [complete_space A]
+[measurable_space A] [borel_space A] [topological_space.second_countable_topology A]
+
+/-- The `limsup` relationship for the spectral radius used to prove `spectrum.gelfand_formula`. -/
+lemma limsup_pow_nnnorm_pow_one_div_le_spectral_radius (a : A) :
+  limsup at_top (λ n : ℕ, ↑∥a ^ n∥₊ ^ (1 / n : ℝ)) ≤ spectral_radius ℂ a :=
+begin
+  refine ennreal.inv_le_inv.mp (le_of_forall_pos_nnreal_lt (λ r r_pos r_lt, _)),
+  simp_rw [inv_limsup, ←one_div],
+  let p : formal_multilinear_series ℂ ℂ A :=
+    λ n, continuous_multilinear_map.mk_pi_field ℂ (fin n) (a ^ n),
+  suffices h : (r : ℝ≥0∞) ≤ p.radius,
+  { convert h,
+    simp only [p.radius_eq_liminf, ←norm_to_nnreal, norm_mk_pi_field],
+    refine congr_arg _ (funext (λ n, congr_arg _ _)),
+    rw [norm_to_nnreal, ennreal.coe_rpow_def (∥a ^ n∥₊) (1 / n : ℝ), if_neg],
+    exact λ ha, by linarith [ha.2, (one_div_nonneg.mpr n.cast_nonneg : 0 ≤ (1 / n : ℝ))], },
+  { have H₁ := (inverse_one_sub_smul_differentiable_on r_lt).has_fpower_series_on_ball r_pos,
+    exact ((inverse_one_sub_smul_has_fpower_series_on_ball ℂ a).exchange_radius H₁).r_le, }
+end
+
+/-- **Gelfand's formula**: Given an element `a : A` of a complex Banach algebra, the
+`spectral_radius` of `a` is the limit of the sequence `∥a ^ n∥₊ ^ (1 / n)` -/
+theorem gelfand_formula (a : A) :
```

Comment:
I think a more descriptive name would be the better option here, like `pow_nnnorm_pow_one_div_tendsto_nhds_spectral_radius` or something like that. Also, I think it would make sense to have a version with the regular norm instead of `nnnorm` in the statement.

Reply:
About the norm version: I think it's not worth the pain. Remember that `spectral_radius` gives you an `ennreal`. Even the statement would be a bit gross I think because it would involve `to_nnreal` followed by the coercion. Or do you see something In missing?